### PR TITLE
openmvg: 2.0 -> unstable-2022-12-30

### PR DESCRIPTION
**Pull request by bouk**
###### Description of changes

The installation instructions recommend latest master. Fixes build on arm and darwin.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/misc/openmvg/default.nix [hunk 1]
```diff
@@ -14,8 +14,8 @@ stdenv.mkDerivation rec {
   src = fetchFromGitHub {
     owner = "openmvg";
     repo = "openmvg";
-    rev = "v${version}";
-    sha256 = "sha256-6F/xUgZpqY+v6CpwTBhIXI4JdT8HVB0P5JzOL66AVd8=";
+    rev = "e1bbfe801986cd7171f36443a1573b0f69f3702d";
```
Comment (AndersonTorres) [inline pkgs/applications/science/misc/openmvg/default.nix:17]: Change version string to `unstable-YYYY-MM-DD` (the date of the rev you are using here).

File: pkgs/applications/science/misc/openmvg/default.nix [hunk 2]
```diff
@@ -40,11 +40,10 @@ stdenv.mkDerivation rec {
   hardeningDisable = [ "all" ];
 
   meta = {
-    broken = (stdenv.isLinux && stdenv.isAarch64);
```
Comment (AndersonTorres) [inline pkgs/applications/science/misc/openmvg/default.nix:43]: Now it is broken on intel-darwin.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/science/misc/openmvg/default.nix (hunk 2 above)
Comment (bouk) [reply pkgs/applications/science/misc/openmvg/default.nix:43]: Looking at the error it seems to be an llvm compiler bug, I tried version 14 and 12 and it doesn't work with either of them, so I'm going to mark it as broken for intel darwin.

**Comment by bouk** [comment]
@AndersonTorres thanks for the review, could you take another look?

**Review by AndersonTorres — APPROVED** [review]
LGTM.
Waiting ofborg.

**Pull request merged**